api/views.py

```diff
@@ -1,27 +1,90 @@
-from django.shortcuts import render
-from django.shortcuts import get_object_or_404
-
-from rest_framework import viewsets
+from rest_framework import status
 from rest_framework.response import Response
+from rest_framework.decorators import api_view
 
 from .serializers import UserSerializer, UserTripSerializer, TripSerializer
 from .models import User, Trip
 
-class UserViewSet(viewsets.ModelViewSet):
-    queryset = User.objects.all()
-    serializer_class = UserSerializer
+@api_view(['GET', 'POST'])
```

Comment:
can you describe what this line is doing either here or on zoom tomorrow? i have _some_ understanding from googling but i'm still a little confused

Reply:
essentially that specifies what type of request is allowed. so for the url this view is associated with, the only allowed methods are GET and POST. The other types of requests return a HTTP error (I believe that it's 405)